Libraries/Storage/AsyncStorage.ios.js

```diff
@@ -27,49 +27,73 @@ var RCTAsyncStorage = RCTAsyncRocksDBStorage || RCTAsyncLocalStorage;
  * operates globally.
  *
  * This JS code is a simple facad over the native iOS implementation to provide
- * a clear JS API, real Error objects, and simple non-multi functions.
+ * a clear JS API, real Error objects, and simple non-multi functions. Each 
+ * method returns a `Promise` object.
  */
 var AsyncStorage = {
   /**
    * Fetches `key` and passes the result to `callback`, along with an `Error` if
-   * there is any.
+   * there is any. Returns a `Promise` object.
    */
   getItem: function(
     key: string,
     callback: (error: ?Error, result: ?string) => void
-  ): void {
-    RCTAsyncStorage.multiGet([key], function(errors, result) {
-      // Unpack result to get value from [[key,value]]
-      var value = (result && result[0] && result[0][1]) ? result[0][1] : null;
-      callback((errors && convertError(errors[0])) || null, value);
+  ): any {
```

Comment:
Done. I hadn't run flow to verify no errors were introduced, but I did prior to my latest commit (`9b571fb0119b3cf1b9dd960fb5875205c33c5c1f `) and received no warnings/errors.